---
title: C28105 |Microsoft Docs
ms.custom: ''
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
caps.latest.revision: 5
author: corob-msft
ms.author: gewarren
manager: ghogen
ms.openlocfilehash: 6b69d84ad48ff83a28dfa2bd172260b7784ba68d
ms.sourcegitcommit: 9ceaf69568d61023868ced59108ae4dd46f720ab
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 10/12/2018
ms.locfileid: "49206774"
---
# <a name="c28105"></a>C28105
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

警告 C28105： 因例外狀況而流失資源  
  
 引發例外狀況時，不會釋放指定的資源。 路徑所指定的陳述式可以引發例外狀況。 這項警告是類似於警告[C28103](../code-quality/c28103.md)，不過在此情況下的相關例外狀況。  
  
## <a name="example"></a>範例  
 下列程式碼範例會產生此警告：  
  
```  
res = KeSaveFloatingPointState(buffer);  
  
res = AllocateResource(Resource);  
char *p2 = new char[10]; // could throw  
  
delete[] p2;  
FreeResource(Resource)  
```  
  
 下列程式碼範例可避免此警告：  
  
```  
res = AllocateResource(Resource);  
char *p2;  
  
try {  
    p2 = new char[10];  
} catch (std::bad_alloc *e) {  
    // just handle the throw  
    ;  
}  
FreeResource(Resource)  
```



